app/js/views/app_view.js

```diff
@@ -1,21 +1,44 @@
 define([
   'jquery',
   'backbone',
-  'text!templates/app_view.ejs'
-], function ($, Backbone, indexTemplate) {
+  'model/todo',
+  'collections/todoList'
+], function ($, Backbone, Todo, TodoList) {
   return Backbone.View.extend({
-    el: this.$('#todo-app'),
+    el: '#todo-app',
 
-    currentDate: function () {
-      return new Date();
+    //events for creating new items
+    events: {
+      'click #addButton': 'createOnButtonClick',
+      'keypress #new-todo': 'createOnEnter'
     },
 
-    render: function () {
-      var compiledTemplate = ejs.render(indexTemplate, {view: this, model: this.model}, {});
+    //bind events on the TodoList collection in order to add items
+    initialize: function () {
+      this.$input = this.$('#new-todo');
+    },
+
+    createTodo: function () {
+      return new Todo({
+        name: this.$input.val().trim(),
+        done: false
+      });
+    },
 
-      this.$el.empty();
-      this.$el.append(compiledTemplate);
-      return this;
+    createOnButtonClick: function () {
```

Comment:
`createOnButton` and `createOnEnter` mostly do the same thing, they also duplicate the validation code, if all o the sudden we  want our input to be 2 chars minimum you will have to update both functions.

If we reorganize the view to use a html `form` you can only listen to form submits (enter and click trigger the same event) and only have one entry point.

Update the implementation in such a way that the add use case is a pipeline of smaller uses cases: validate | addToCollection.